### DISCUSSION TITLE
How to use withDependencies() when appending to NavigationStack path

### DISCUSSION THREAD

**Discussion by alexbird** [pointfreeco/swift-composable-architecture#3150] [General]
It seems that the Store needs to be created inside a withDependencies context. Stack navigation seems to have two points where we construct things, but the actual Store creation happens within the TCA library code. Am I missing something?

We construct:
- The State, e.g. 
```swift
case .goToABCButtonTapped:
        state.path.append(.screenA(ScreenA.State()))
        return .none
``` 
- The View, e.g.
```swift
    } destination: { store in
      switch store.case {
      case let .screenA(store):
        ScreenAView(store: store)
      }
    }
```

...but the `Store` has been constructed between these definitions, and wrapping neither of these in `withDependencies` seems to override anything.

**Comment by mbrandonw** [comment]
Hi @alexbird, in TCA you override dependencies with the [`.dependency`](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/reduce/dependency(_:_:)) reducer operator (or what of the related APIs). 

Can you give more details of what dependency you are wanting to override and where?

**Reply by alexbird** [reply]
My scenario is a feature displaying various live information on a map, so it has dependencies I've created for API access and a stream of user locations. I want to navigate to this feature from a testbed/demo root feature, which can select either real data or canned data for demonstration purposes. 
Would you suggest the way to do this  embedding the feature in new reducer/s which uses the `.dependency` operator, and adding Path case/s for these reducers?

**Reply by alexbird** [reply]
That does the job, thanks.

I also needed (?) to create a minimal View to embed the feature in, a la:
```swift
struct CannedTripMapView: View {
    let store: StoreOf<CannedTripMap>
    
    var body: some View {
        TripMapView(store: store.scope(state: \.feature, action: \.feature))
    }
}
```

It would be handy if I could inject dependencies into this wrapper, but I cannot see how to use other parts of its state within the `Scope()` operator. Perhaps this is crazy talk?
